torch/_inductor/select_algorithm.py

```diff
@@ -752,7 +759,9 @@ def load_input(
         """
 
         input_node = self.named_input_nodes[input_name]
-        self.prologue_supported_inputs.add(input_node.get_name())
+        if not self.all_inputs_loaded:
+            self.prologue_supported_inputs.add(input_node.get_name())
+
```

Comment:
This is an OrderedSet, so we dont need this check. If we're going to do this we should insert that its already in the set